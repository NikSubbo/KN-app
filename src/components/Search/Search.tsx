import { FC } from 'react';
import { ReactComponent as SearchIcon } from '../../assets/search.svg';

type Props = {
  search: string;
  handleSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
};

export const Search: FC<Props> = ({ search, handleSearchChange }) => {
  return (
    <div className="flex bg-smoke border-1 border-lightGray rounded-sm p-1.5">
      <SearchIcon />
      <input
        type="text"
        value={search}
        onChange={handleSearchChange}
        placeholder="Type here..."
        className="bg-smoke ml-1 pl-2 focus:outline-none w-full"
      />
    </div>
  );
};

export default Search;
