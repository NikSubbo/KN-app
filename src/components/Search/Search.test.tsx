import { render } from '@testing-library/react';
import Search from './Search';

describe('Search', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Search search="" handleSearchChange={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
