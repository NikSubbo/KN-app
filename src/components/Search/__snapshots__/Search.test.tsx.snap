// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div>
  <div
    class="flex bg-smoke border-1 border-lightGray rounded-sm p-1.5"
  >
    <svg>
      search.svg
    </svg>
    <input
      class="bg-smoke ml-1 pl-2 focus:outline-none w-full"
      placeholder="Type here..."
      type="text"
      value=""
    />
  </div>
</div>
`;
